File: packages/gbnf-experiment/data/20260909T123239Z_5982b634/ported_implementation/src/grammar-graph/get-input-as-code-points.ts
import type { ValidInput } from './types.js';

export const getCodePoint = (char: string): number => {
  const codePoint = char.codePointAt(0);
  if (codePoint === undefined) {
    throw new Error(`Could not get code point for character: ${char}`);
  }
  return codePoint;
};

export const getInputAsCodePoints = (src: ValidInput): number[] => {
  if (typeof src === 'number') {
    return [src];
  }

  if (Array.isArray(src)) {
    for (const codePoint of src) {
      if (!Number.isInteger(codePoint)) {
        throw new Error(`code_point must be an integer for ${codePoint} if src is a list`);
      }
    }
    return src;
  }

  if (typeof src === 'string') {
    return [...src].map(getCodePoint);
  }

  throw new Error(`Invalid input type: ${typeof src}`);
};

File: proposed-projects/round-trip-experiment/reverse/20260909T151143Z_170b737d/20260910T085124Z_7f672549/ported_implementation/gbnf/grammar_graph/get_input_as_code_points.py
from typing import List

from .grammar_graph_types import ValidInput


def get_code_point(char: str) -> int:
    if not char:
        raise ValueError(f"Could not get code point for character: {char}")
    return ord(char)


def get_input_as_code_points(src: ValidInput) -> List[int]:
    if isinstance(src, bool):
        raise ValueError(f"Invalid input type: {type(src).__name__}")

    if isinstance(src, int):
        return [src]

    if isinstance(src, list):
        for c in src:
            if not isinstance(c, int) or isinstance(c, bool):
                raise ValueError(
                    f"code_point must be an integer for {c} if src is a list"
                )
        return src

    if isinstance(src, str):
        return [get_code_point(s) for s in src]

    raise ValueError(f"Invalid input type: {type(src).__name__}")
